### ISSUE TITLE
svelte/no-useless-mustaches is too restrictive

### ISSUE THREAD

**Issue by gyzerok** [sveltejs/eslint-plugin-svelte#1179]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.23.0

### What version of `eslint-plugin-svelte` are you using?

3.4.1

### What did you do?

Sometimes you need to explicitly and visible add whitespace in your markup. I am coming from React and there it's a common pattern to use `{' '}` in such places. It seems like there is no other way in Svelte to achieve this, so I think eslint plugin shouldn't be that restrictive.

The bigger question here is - is it the only case or should this rule skip mustaches inside tags altogether and validate only properties such as `foo={'bar'}`?

<details>
<summary>Configuration</summary>

```
export default ts.config(
	includeIgnoreFile(gitignorePath),
	js.configs.recommended,
	...ts.configs.recommended,
	...svelte.configs.recommended,
	prettier,
	...svelte.configs.prettier,
	{
		languageOptions: {
			globals: { ...globals.browser, ...globals.node }
		},
		rules: { 'no-undef': 'off' }
	},
	{
		files: ['**/*.svelte', '**/*.svelte.ts', '**/*.svelte.js'],
		ignores: ['eslint.config.js', 'svelte.config.js'],
		languageOptions: {
			parserOptions: {
				projectService: true,
				extraFileExtensions: ['.svelte'],
				parser: ts.parser,
				svelteConfig
			}
		}
	}
);
```
</details>

```svelte
<div>
  foo
  {#if something}
    {' '}bar
  {/if}
</div>
```

### What did you expect to happen?

No linter errors

### What actually happened?

`Unexpected mustache interpolation with a string literal value  svelte/no-useless-mustaches`

### Link to **GitHub Repo** with Minimal Reproducible Example

Look for the error for `Bug2` component: https://github.com/gyzerok/svelte-eslint-bug

**Comment by baseballyama** [comment]
The rendered result looks the same regardless of whether parentheses are used.

https://svelte.dev/playground/hello-world?version=5.25.5#H4sIAAAAAAAAE42MMWoDMRBFrzL8FE5gsXuxMaTLHawUsjVKBuTRshrbCYvuHmQT3LhI9eHx3l-g4chweOecC13KnCM9cxTj-IIBSTJXuN0C-5m61wGGv-ptmtb1zNk624fKj_ihqLFahcNYD7NMtvXqLbNRLUe2L9FPeiWP4OF13NwdHaOct16JUil9lidJ96h1RLSsaNX2Yb4KG0mtn9zCfz48ijHA-NvgbD5x-xhgQfJFNMKlkCu3X-D9Tig7AQAA

**Comment by gyzerok** [comment]
@baseballyama yeah, noticed this, but am not sure why it is happening. Because if I move cursor around in my IDE it looks like there is no space there. So in this specific case I think it's better to not rely on something "invisible", that's why I am keep on using explicit space there

**Comment by gyzerok** [comment]
While migrating to never plugin version I indeed swapped some places with `&nbsp;`, but it's not suitable for all the places - in some places space has to continue be breakable

**Comment by baseballyama** [comment]
In HTML, consecutive spaces are treated as a single space, so still I don't get what is the issue.
https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace
